### PR TITLE
Various fixes and updates

### PR DESCRIPTION
**Pull request by Bogdanp**
* adds `elm-interactive-mode` to replace `elm-repl`
* rewrites `elm-compile` with compilation buffer support (along with error regexps to make navigation a breeze!)
* rewrites `elm-preview` to use `elm-reactor`
* adds `exposing` keyword
* adds dependencies on `f`, `s` and `let-alist` to cut down on boilerplate code
* adds optional dependency on haskell mode and experimental "load decl. into repl" functionality (`C-c C-e`)
* cleans up some of the elisp

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,34 +1,37 @@
...
-1. Syntax highlighting
-2. Intelligent indentation
-3. Integration with [elm-repl](https://github.com/elm-lang/elm-repl)
+1. Syntax highlighting.
+1. Intelligent indentation.
+1. Integration with [elm-repl](https://github.com/elm-lang/elm-repl).
 
 ## Installation
 
-If your Emacs has `package.el` (which is automatically the case for Emacs >= 24), you can install `elm-mode` from the package in [MELPA](http://melpa.milkbox.net/).
```
Comment (jcollard) [inline README.md]: Why remove instructions on using MELPA?

File: README.md [hunk 2]
```diff
@@ -1,34 +1,37 @@
...
 
-```
-    (setenv "PATH" (concat (getenv "PATH") ":~/.cabal/bin"))
-    (setq exec-path (append exec-path '("~/.cabal/bin")))
-```
+  <dt><code>C-c C-c</code></dt>
+  <dd>Compile the current buffer.</dd>
+
+  <dt><code>C-c C-n</code></dt>
+  <dd>Preview the current buffer in a browser.</dd>
```
Comment (jcollard) [inline README.md]: Might want to specifically say that this compiles locally and launches the browser. This should really be updated to use elm-reactor.

File: README.md (hunk 1 above)
Comment (Bogdanp) [inline README.md]: That was mainly to avoid confusion for people who end up on my repo. I'll revert that hunk tomorrow and update the PR.

**Comment by Bogdanp** [comment]
@jcollard do let me know if you find the lack of cross-variant compatibility an issue so I can close the PR and stop the e-mail spam I'm sure you're getting due to my commits. :)

**Comment by jcollard** [comment]
No worries. I really do appreciate the help. I actually only use emacs as well so I can't test it either. We'll just have to see if someone has issues.

Thanks for the help!

**Pull request merged**